unit-tests/src/org/commcare/android/tests/formsave/FormRecordProcessingTest.java

```diff
@@ -0,0 +1,175 @@
+import org.commcare.android.tests.queries.CaseDbQueryTest;
+import org.commcare.android.util.TestAppInstaller;
+import org.commcare.android.util.TestUtils;
+import org.commcare.dalvik.BuildConfig;
+import org.commcare.dalvik.R;
+import org.commcare.dalvik.activities.CommCareHomeActivity;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.session.CommCareSession;
+import org.commcare.session.SessionNavigator;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.reference.ReferenceManager;
+import org.javarosa.core.reference.ResourceReferenceFactory;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.views.ODKView;
+import org.odk.collect.android.widgets.IntegerWidget;
+import org.robolectric.Robolectric;
+import org.robolectric.Shadows;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowActivity;
+import org.robolectric.shadows.ShadowEnvironment;
+
+import java.util.Date;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+
+/**
+ * Form Record processing / form save related tests
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+@Config(application = CommCareApplication.class,
+        constants = BuildConfig.class)
+@RunWith(CommCareTestRunner.class)
+public class FormRecordProcessingTest {
+    private static final String TAG = FormRecordProcessingTest.class.getSimpleName();
+
+    @Before
+    public void setup() {
+        // needed to resolve "jr://resource" type references
+        ReferenceManager._().addReferenceFactory(new ResourceReferenceFactory());
+
+        TestUtils.initializeStaticTestStorage();
+        TestAppInstaller.setupPrototypeFactory();
+
+        TestAppInstaller appTestInstaller =
+                new TestAppInstaller("jr://resource/commcare-apps/form_save_regressions/profile.ccpr",
+                        "test", "123");
+        appTestInstaller.installAppAndLogin();
+        ShadowEnvironment.setExternalStorageState(Environment.MEDIA_MOUNTED);
+    }
+
+    @Test
+    public void testCaseIndexUpdateDuringFormSave() {
+        TestUtils.processResourceTransactionIntoAppDb("/inputs/case_create_for_index_table_test.xml");
+
+        EvaluationContext ec = TestUtils.getEvaluationContextWithAndroidIIF();
+
+        CaseDbQueryTest.evaluate("instance('casedb')/casedb/case[@case_type = 'followup'][index/parent = 'worker_one']/@case_id",
+                "constant_id", ec);
+
+        fillOutFormWithCaseUpdate();
+
+        // TODO PLM: I would expect the following evaluation to not result in 'constant_id'
+        CaseDbQueryTest.evaluate("instance('casedb')/casedb/case[@case_type = 'followup'][index/parent = 'worker_one']/@case_id",
+                "constant_id", ec);
```

Comment:
This test saves a form that updates the case parent from `worker_one` to `worker_two`, hence I would expect this assert to fail, because the shouldn't a case only have one parent? And in this case it should be `worker_two`, as is asserted right below.

I think this exemplifies a bug in the system.